llama_index/postprocessor/rankGPT_rerank.py

```diff
@@ -2,68 +2,38 @@
 from typing import Any, Dict, List, Optional, Sequence
 
 from llama_index.bridge.pydantic import Field
-from llama_index.llms import ChatMessage, ChatResponse, LiteLLM, OpenAI
-from llama_index.llms.openai_utils import (
-    openai_modelname_to_contextsize,
-)
+from llama_index.llms import ChatMessage, ChatResponse, OpenAI
+from llama_index.llms.llm import LLM
 from llama_index.postprocessor.types import BaseNodePostprocessor
 from llama_index.schema import NodeWithScore, QueryBundle
-from llama_index.utilities.token_counting import TokenCounter
 from llama_index.utils import print_text
 
-DEFAULT_OPENAI_MODEL = "gpt-3.5-turbo-0301"
-
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.WARNING)
 
 
 class RankGPTRerank(BaseNodePostprocessor):
     """RankGPT-based reranker."""
 
-    top_n: int = Field(description="Top N nodes to return from reranking.")
-    model: str = Field(
-        default=DEFAULT_OPENAI_MODEL, description="The LLM model to use."
-    )
-    temperature: float = Field(
-        default=0.0,
-        description="The temperature to use during generation.",
-        gte=0.0,
-        lte=1.0,
-    )
-    api_key: str = Field(default=None, description="The LLM API key.", exclude=True)
-    context_size: int = Field(
-        default=3000,
-        description="The context size to use for LLM models. Need to set for non-GPT models.",
-        gte=0,
-        lte=2048,
+    top_n: int = Field(default=5, description="Top N nodes to return from reranking.")
+    llm: LLM = Field(
+        default=OpenAI("gpt-3.5-turbo"), description="LLM to use for rankGPT"
```

Comment:
To avoid circular import errors, make this a default factory instead
`default_factory=OpenAI`